docs/index.md

```diff
@@ -0,0 +1,39 @@
+# Calliope: energy system modelling made simple
+
+!!! note
```

Comment:
This doesn't seem necessary. RTD allows for version warnings: https://docs.readthedocs.io/en/stable/versions.html#version-warning

Then you don't need to track version in the mkdocs config and don't need to worry about making it dynamic.